tests/RobotFramework/tests/tedge/apt_plugin.robot

```diff
@@ -0,0 +1,69 @@
+*** Settings ***
+Documentation       Two settings have been specifically added to tedge-apt-plugin.
+...    tedge-apt-plugin list --name $PATTERN lists the packages which name matches the regular expression $PATTERN.
+...    tedge-apt-plugin list --maintainer $PATTERN lists the packages which maintainer matches the regular expression $PATTERN.
+...    If both --name and --maintainer patterns are provided, tedge-apt-plugin lists packages which either the name or the maintainer matches.
+...    These two settings can stored in /etc/tedge/tedge_config.toml in an apt table.
+...    If stored in /etc/tedge/tedge_config.toml, then the apt.name and apt.maintainer settings are used to filter packages listed to the cloud.
+...    No support is provided by tedge config so store apt.name and apt.maintainer.
+
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Setup
+Suite Teardown      Get Logs
+
+
+*** Test Cases ***
+
+
+tedge-apt-plugin list
+    ${apt_list}    Execute Command    /etc/tedge/sm-plugins/apt list
+    Should Contain    ${apt_list}    apt
+    # Should Contain    ${apt_list}    wget
+    Should Contain    ${apt_list}    dpkg
+
+tedge-apt-plugin list --name
+    ${apt_list_name}    Execute Command    /etc/tedge/sm-plugins/apt list --name '(c8y-).*'
+    Should Contain    ${apt_list_name}    c8y-configuration-plugin
+    Should Contain    ${apt_list_name}    c8y-firmware-plugin
+    Should Contain    ${apt_list_name}    c8y-log-plugin
+    Should Contain    ${apt_list_name}    c8y-remote-access-plugin
+    Should Not Contain    ${apt_list_name}    tedge
+    Should Not Contain    ${apt_list_name}    tedge-agent
+    Should Not Contain    ${apt_list_name}    tedge-apt-plugin
+    Should Not Contain    ${apt_list_name}    tedge-mapper
+    Should Not Contain    ${apt_list_name}    tedge-watchdog
+
+
+tedge-apt-plugin list --maintainer
+    ${apt_workaround}    Set Variable    apt    
+    ${apt_list_maintainer}    Execute Command    /etc/tedge/sm-plugins/apt list --maintainer '.*(thin-edge.io).*'
+    Should Contain    ${apt_list_maintainer}    c8y-configuration-plugin
+    Should Contain    ${apt_list_maintainer}    c8y-firmware-plugin
+    Should Contain    ${apt_list_maintainer}    c8y-log-plugin
+    Should Contain    ${apt_list_maintainer}    c8y-remote-access-plugin
+    Should Contain    ${apt_list_maintainer}    tedge
+    Should Contain    ${apt_list_maintainer}    tedge-agent
+    Should Contain    ${apt_list_maintainer}    tedge-apt-plugin
+    Should Contain    ${apt_list_maintainer}    tedge-mapper
+    Should Contain    ${apt_list_maintainer}    tedge-watchdog
+    Should Not Contain    ${apt_list_maintainer}    ${apt_workaround}\t
```

Comment:
I don't think `\t` will be interpreted as a TAB character...I think it is 'working' because the ${apt_workaround}\t is not present (which is isn't).

And I would not call it a 'workaround' as it is not a hack, it is just being more specific about what string you are looking for to prevent false positives.

```suggestion
    # Note: use a trailing ${\t} to prevent matching against `tedge-apt-plugin`
    Should Not Contain    ${apt_list_maintainer}    apt${\t}
```

Reply:
You can verify if `${\t}` is the correct RobotFramework syntax by using it in the `Should Contain` example (just for local verification first)

For example this should match

```
Should Contain    ${apt_list_maintainer}    tedge-watchdog${\t}
```

Reply:
Just to close this comment. The original usage of `\t` was correct. It is only the newline character which supports being defined as `${\n}`, a tab character is correctly interpreted by the RobotFramework as a TAB character when used within any string.